app/helpers/date_validation_helper.rb

```diff
@@ -18,6 +18,8 @@ def valid_or_invalid_date(year, month)
   end
 
   def start_date_before_end_date
+    return unless start_date.is_a?(Date) && end_date.is_a?(Date)
```

Comment:
Right, so if this is a `Date` rather than a `Struct` it means we've successfully parsed it? I'd almost go so far as to name the Struct `IncompleteDate` or sth  in `DateValidationHelper`?

Reply:
The models only run this check if there are no errors on the `start_date` or `end_date` fields already. As we can potentially have an empty start_date struct and no errors due to the start_date not being mandatory. The same applies for end_date. 

https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/forms/candidate_interface/work_experience_form.rb#L19